core-skills/c-sharp-and-dotnet/README.md

```diff
@@ -0,0 +1,84 @@
+# C# and .NET Core
+
+## Who is this skill for?
+
+This skill is intended for developers who have already completed the TDD core skill, and who have not worked with .NET previously (or want to brush up) and are likely to join a delivery team that is working in C#.
+Through this course, we'll be making some assumptions that you're already familiar with at least one programming language (static typed or dynamic).
+
+## Summary
+### Mantis Shrimp
+1. Can set up a working project in .NET using an IDE
+2. Can install and manage dependencies for the project
+3. Can create and run unit tests using a test framework
+4. Understands basic concepts of the C# language and ecosystem
+
+### Goat
+1. Understands some deeper object oriented programming concepts
+2. Able to demonstrate common design patterns implemented using C# language features
+3. Understands how to build API functionality using common frameworks (MVC, EF)
+4. Understands how to use the dotnet command line utilities to perform common tasks
+5. Can effectively debug using the interactive debugger (IDE)
+
+### Eagle
+1. Demonstrate an understanding of functional programming in C#
+2. Understanding of the .NET virtual machine
+3. Understanding the impact of choices on performance
+4. Demonstrate ability to combine understanding of the above into a working API in C# with .NET
+
+## Resources
+Coming soon
+
+## Mark Scheme
+### Mantis Shrimp
+
+- Can explain the difference between .NET, .NET Core and Framework
+- Can explain what the .NET Standard is 
+- Can explain the difference between C#, Visual Basic and F#
+- Can explain what assemblies and artefacts are
+- Can explain the concept of classes with examples of usage
+- Can explain the concept of interfaces with examples of usage
+- Can explain the concept of inheritance with examples of usage
+
+- Can create a new project in an IDE
```

Comment:
Considering Microsoft is improving integrations with things such as VSCode, and the `dotnet` CLI allows you to do a good amount. Do we need to necessitate an IDE here? (Whatever the definition of an IDE is these days) 

Or is the aim to actually require the use of an IDE (given there's a couple other references to IDEs in the marking)

Reply:
I think IDE development is important for .NET specifically, and you should be able to at least convincingly navigate one that is suitable. Not necessarily restring people to Rider/VS/VSCode etc, but a few basic IDE actions, project setup (which is still very much IDE based due to auto generated files and dependancy management) and refactoring/debugging proficiency are IMO necessary for being delivery capable.

Reply:
VSCode from what I remember doesn't necessarily have great refactoring tools - does this mean you wouldn't be able to pass that marking criteria? Project setup isn't too bad with the `dotnet new` command from what I found - it's a pretty powerful tool and it's what IDEs use also.

I feel like you could mention IDEs as an important thing in the resources/preamble but unless we're starting to enforce text editors on developers as a company I feel 50/50 on including IDE specific stuff as a requirement. Not sure the right way to go

I'd be interested to know others thoughts on it

Reply:
VSCode has refactoring tools that do actually work, and given the nature of a .NET codebase you _want_ to be familiar with the quick options for doing things such as renaming all instances and moving namespaces etc. I think knowing enough IDE to be dangerous is actually a requirement for delivering, and if your IDE of choice forces you into using `dotnet new`, you can still pass this and get a sticker, as long as you know what you're about 🤷‍♂

Reply:
Sounds like we could simplify it to
```suggestion
- Can create a new C# project
```